rmw_zenoh_cpp/src/detail/attachment_helpers.hpp

```diff
@@ -23,11 +23,47 @@
 
 namespace rmw_zenoh_cpp
 {
+
+class attachement_data_t final
+{
+public:
```

Comment:
I think this would introduce lots of complexity since we need to do some serializations in attachment_helpers.cpp which need to access the values. To make them private, we need to provide either public serialization functions or public get functions (including returning an array pointer of a known sized array `source_gid[RMW_GID_STORAGE_SIZE]`)